src/codecs/decoding/format/syslog.rs

```diff
@@ -69,7 +95,7 @@ fn insert_fields_from_syslog(event: &mut Event, parsed: Message<&str>) {
         );
     }
     if let Some(host) = parsed.hostname {
-        log.insert("hostname", host.to_string());
+        log.insert(log_schema().host_key(), host.to_string());
```

Comment:
This _can_ be a breaking change, for operators that changed the default key for the "host" schema field name.

I can keep this in, or revert this, if we prefer not to make this change (it isn't needed for schema work, but seemed like an oversight in the initial implementation).

cc @jszwedko

Reply:
I reverted this, as it broke existing tests, and the implementation of those tests _seem_ to apply that this is _by design_. We can revisit this in a follow-up PR, but since it's not directly related to the purpose of this PR, and it's a potential breaking change regardless, I'm backing out of this change here.